From 37ab8ed18f35591f2bd99043f12c06d98b4527db Mon Sep 17 00:00:00 2001
From: Michael Webster <miketwebster@gmail.com>
Date: Thu, 7 Dec 2023 09:05:41 -0500
Subject: [PATCH] cinnamon-screensaver-main.py: Fix application theme provider.

We relied on the return value from Gtk.CssProvider.load_from_data
method to determine whether or not to apply the fallback theme.

The return value itself was deprecated, but still works in Gtk3.
Unfortunately a change was recently made in pygobject that overrides
this method, and since the return value is removed in Gtk4, it
was removed from this override, so is always interpreted as False.

Use try/except instead, as that method can still raise an error
upon failure.

Fixes #446

Ref:
https://gitlab.gnome.org/GNOME/pygobject/-/merge_requests/231
---
 src/cinnamon-screensaver-main.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/cinnamon-screensaver-main.py b/src/cinnamon-screensaver-main.py
index 05b727c..5c5e486 100755
--- a/src/cinnamon-screensaver-main.py
+++ b/src/cinnamon-screensaver-main.py
@@ -139,9 +139,12 @@ def do_style_overrides(self):
 
             fallback_prov = Gtk.CssProvider()
 
-            if fallback_prov.load_from_data(fallback_css.encode()):
+            try:
+                fallback_prov.load_from_data(fallback_css.encode())
                 Gtk.StyleContext.add_provider_for_screen (Gdk.Screen.get_default(), fallback_prov, 600)
                 Gtk.StyleContext.reset_widgets(Gdk.Screen.get_default())
+            except Exception as e:
+                print("Could not parse fallback css: %s" % str(e))
 
 if __name__ == "__main__":
     setproctitle.setproctitle('cinnamon-screensaver')
